src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearchindexer/repository/PrisonerRepository.kt

```diff
@@ -0,0 +1,17 @@
+package uk.gov.justice.digital.hmpps.prisonersearchindexer.repository
+
+import org.springframework.data.elasticsearch.repository.ElasticsearchRepository
+import org.springframework.data.repository.NoRepositoryBean
+import org.springframework.stereotype.Repository
+import uk.gov.justice.digital.hmpps.prisonersearchindexer.model.Prisoner
+import uk.gov.justice.digital.hmpps.prisonersearchindexer.model.PrisonerA
+import uk.gov.justice.digital.hmpps.prisonersearchindexer.model.PrisonerB
+
+@NoRepositoryBean
+interface PrisonerRepository<T : Prisoner, String> : ElasticsearchRepository<T, String>
+
+@Repository
+interface PrisonerARepository : PrisonerRepository<PrisonerA, String>
+
+@Repository
+interface PrisonerBRepository : PrisonerRepository<PrisonerB, String>
```

Comment:
Similar with the A and B repositories - I think we preferred the method used by probation search indexer where we created an [alias for each index](https://github.com/ministryofjustice/probation-offender-search-indexer/blob/main/src/main/kotlin/uk/gov/justice/digital/hmpps/indexer/repository/OffenderRepository.kt#L58)

Reply:
cool, will pick up next